### ISSUE TITLE
Creating a custom type for validation

### ISSUE THREAD

**Issue by noah-goodrich** [mtth/avsc#214]
We are looking to use your library for schema validation and it would be nice to be able to declare custom types with custom validators. 

Is this possible? 

For example, we need to validate ISO 8601 date strings and email address strings.

**Comment by mtth** [comment]
Yes, it's possible using [logical types](https://github.com/mtth/avsc/wiki/Advanced-usage#logical-types). [This answer](https://stackoverflow.com/a/37282006/1062617) uses one to validate strings (the example predates ES2015 so the code would look slightly simpler now).
